Quick Start: New visuals v2

momo-ozawa opened the pull request:
Part of #18621

## Description
- Updated the Quick Start card design to match the original proposed design
- Design ref: pbArwn-4wS-p2

## Context
- We initially decided to modify the original proposed design to a simpler design (Ref: p1652976178640989/1652793339.295619-slack-C027K4MNPGQ)
- Since there was some time left over in the sprint, I looked into matching the original proposed design (Ref: p1653496943994179/1652793339.295619-slack-C027K4MNPGQ)

## How to test

1. Set the initial screen to "Home" via the App Settings
2. Enable Quick Start for Existing Users via the debug menu
3. ✅ Verify: The Quick Start card design matches the original proposed design
4. Tap the ellipsis button then tap "Remove Next Steps"
5. ✅ Verify: The Quick Start card is removed from the dashboard
6. Set the initial screen to "Menu" via the App Settings
7. Enable Quick Start for Existing Users via the debug menu
8. ✅ Verify: The Quick Start card design matches the original proposed design -- and that the leading / trailing margins of content inside the Quick Start card is the same (i.e. no gap on the trailing edge)
9. Switch to a site that has no posts
10. ✅ Verify: The No Posts card design remains the same as before (i.e. no gap on the trailing edge)

![Simulator Screen Shot - iPhone 13 - 2022-05-26 at 16 27 05](https://user-images.githubusercontent.com/6711616/170523006-7bed0979-1925-4dd6-b114-ac0f2fd8815f.png) | ![Simulator Screen Shot - iPhone 13 - 2022-05-26 at 16 27 25](https://user-images.githubusercontent.com/6711616/170523018-56b56b31-c7c7-457e-8f67-919387c41171.png) | ![Simulator Screen Shot - iPhone 13 - 2022-05-26 at 16 27 38](https://user-images.githubusercontent.com/6711616/170523029-df2c40c7-76c2-403d-8aab-06542c68f571.png) | ![Simulator Screen Shot - iPhone 13 - 2022-05-26 at 16 27 44](https://user-images.githubusercontent.com/6711616/170523033-d93ad215-9a56-494b-baca-59dc7cddde24.png)
-- | -- | -- | --



## Regression Notes
1. Potential unintended areas of impact
- Dashboard cards without headers

9. What I did to test those areas of impact (or what existing automated tests I relied on)
- Tested the steps above

11. What automated tests I added (or what prevented me from doing so)
- n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Commit:
Refactor: hide header for quick start card

Commit:
Refactor: extract stackview constarint configuration into separate methods

Commit:
Update: release notes

osullivanchris reviewed (approved):
Hey sorry I installed the build yesterday but forgot to reply. It looks great. Nice work figuring out a way to do the original design!

leandroalonso reviewed (approved):
@momo-ozawa looks great!

One small thing — not related to your PR:

When the tasks screen is shown (after tapping the card), this warning is being triggered:

```
2022-05-27 09:31:52.066539-0300 WordPress[7576:46691] [TableView] Changing the background color of UITableViewHeaderFooterView is not supported. Use the background view configuration instead.
```

Commit:
Fix: UITableViewHeaderFooterView background color warning

Commit:
Fix: update source rect for QS card ellipsis button

Commit:
Merge branch 'trunk' into task/qs-card-visuals-v2

momo-ozawa commented:
> When the tasks screen is shown (after tapping the card), this warning is being triggered:

@leandroalonso Good catch! Fixed! eaefb7b

FYI I also fixed an iPad bug -- I updated the source rect to point to the `buttonContainerStackView` 70f935a

Before | After 
-- | --
![Simulator Screen Shot - iPad Air (4th generation) - 2022-05-27 at 14 40 34](https://user-images.githubusercontent.com/6711616/170712838-c2b11cf6-46be-4714-acba-381c5dd80c74.png) | ![Simulator Screen Shot - iPad Air (4th generation) - 2022-05-27 at 14 46 38](https://user-images.githubusercontent.com/6711616/170712848-3dd5b16e-cdb4-4e96-9e37-40b7655150f5.png)
